src/substantial/src/converters.rs

```diff
@@ -78,13 +79,126 @@ pub struct Operation {
 /// Each operation is produced from the workflow execution
 #[derive(Serialize, Deserialize, Clone, Debug)]
 pub struct Run {
+    pub id: u32,
     pub run_id: String,
     pub operations: Vec<Operation>,
 }
 
+#[derive(Debug)]
+pub enum Interupt {
+    Sleep,
+    Saveretry,
+    WaitReceiveEvent,
+    WaitHandleEvent,
+    WaitEnsureValue,
+}
+
+impl Interupt {
+    const PREFIX: &'static str = "SUBSTANTIAL_INTERRUPT_";
+}
+
+impl fmt::Display for Interupt {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        let variant = match self {
+            Self::Sleep => "SLEEP",
+            Self::Saveretry => "SAVE_RETRY",
+            Self::WaitReceiveEvent => "WAIT_RECEIVE_EVENT",
+            Self::WaitHandleEvent => "WAIT_HANDLE_EVENT",
+            Self::WaitEnsureValue => "WAIT_ENSURE_VALUE",
+        };
+        write!(f, "{}{:?}", Self::PREFIX, variant)
+    }
+}
+
+impl std::error::Error for Interupt {}
+
+pub enum Strategy {
+    Linear,
+}
+
+pub struct Retry {
+    pub strategy: Option<String>,
+    pub min_backoff_ms: i32,
+    pub max_backoff_ms: i32,
+    pub max_retries: i32,
+}
+
```

Comment:
move into another file maybe? `converters.rs` should only contain conversion code imo.